### PR TITLE
Attempt to make compatible with Julia nightly

### PR DESCRIPTION
**Pull request by pat-alt**
Hi there, 

thanks for this great package. I've added it as a dependency for [CounterfactualExplanations.jl](https://github.com/pat-alt/CounterfactualExplanations.jl) and it saved me a lot of pain the other day 😆. I noticed that the nightly version no longer passes tests though - see the issue in my own package CI [here](https://github.com/pat-alt/CounterfactualExplanations.jl/runs/7577793825?check_suite_focus=true#step:6:376) and related warning [here](https://github.com/pat-alt/CounterfactualExplanations.jl/runs/7577793825?check_suite_focus=true#step:6:373).

After replacing all `Slices` by `JuliennedArrays.Slices` the CI for nightly now passed without errors on my fork, but an error that I don't understand came up for Julia 1.3. Contributing this anyway in the hope that perhaps you know what to do.

Thanks!

### CODE REVIEW

**Comment by mcabbott** [comment]
Great, thanks for having a look. 

Can you try deleting the manifest here? I think this should not be checked in. And it might be that this is what's causing the failure on 1.3. 

If that doesn't work, simply dropping Julia 1.3 is totally fine, it's ancient history.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         version:
-          - '1.3'
+          - '1.7'
```
Comment (mcabbott) [inline .github/workflows/ci.yaml]:
```suggestion
          - '1.3'
          - '1.6'  # LTS 
```

File: Project.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ StaticArrays = "1.0"
 Tracker = "0.2"
 Zygote = "0.6"
 ZygoteRules = "0.2"
-julia = "1.3"
+julia = "1.7"
```
Comment (mcabbott) [inline Project.toml]:
```suggestion
julia = "1.3"
```

**Comment by pat-alt** [comment]
Have moved to 1.7 and also bumped up the package version 👍🏽

**Review by mcabbott — COMMENTED** [review]
The reason CI doesn't run is that it seems to be set up to require my approval first, sorry. Reverting the version change to see whether it works without the manifest first...

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ jobs:
       matrix:
         version:
           - '1.3'
```
Comment (mcabbott) [inline .github/workflows/ci.yaml]:
```suggestion
```

**Review by mcabbott — COMMENTED** [review]
ok, let's give up on 1.3. But I think the lower bound should be 1.6.

**Comment by mcabbott** [comment]
Thanks!

**Pull request merged**